TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/ScoringAlgorithm.java

```diff
@@ -0,0 +1,92 @@
+package org.firstinspires.ftc.teamcode.auton;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.Servo;
+
+import org.firstinspires.ftc.teamcode.subsystems.hardware.Hardware;
+import org.firstinspires.ftc.teamcode.subsystems.hardware.HardwareElement;
+
+@Autonomous(name = "Scoring Algorithm", group = "auton")
+public class ScoringAlgorithm extends OpMode {
+
+    private Hardware robot;
+    // TODO: Get values for variables.
+    boolean pixelsInClaw = false;
+    double armTargetPos = 0.0;
+    double viperSlideTargetPos = 0.0;
+    double jointTargetPos = 0.0;
+
+    public void init() {
+        telemetry.addData("Status", "Initializing");
+
+        robot = new Hardware(hardwareMap, telemetry);
+        // Introducing servos.
+        robot.introduce(new HardwareElement<>(Servo.class, hardwareMap, "clawServo"));
+        robot.introduce(new HardwareElement<>(Servo.class, hardwareMap, "jointServo"));
+        // Introducing motors and encoders.
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "armMotor"));
+        robot.<DcMotor>get("armMotor").setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "rightViperSlideMotor"));
+        robot.<DcMotor>get("rightViperSlideMotor").setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "leftViperSlideMotor"));
+        robot.<DcMotor>get("leftViperSlideMotor").setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+
+        telemetry.addData("Status", "Initialized");
+    }
+
+    private void scoring() {
+        telemetry.addData("Scoring Status", "Started");
+
+        setViperSlidePos();
+
+        setArmPos();
+
+        setJointPos();
+
+        while (pixelsInClaw) {
+            robot.<Servo>get("clawServo").setPosition(robot.<Servo>get("clawServo").getPosition() - 0.125);
+            wait();
```

Comment:
not sure why we're using `wait()` here. That's a multithreading method- you would need a separate threat to run `notify()`. Look up the `synchronized` keyword if you're not sure what that is :)